learning/step_by_step/godot_design_philosophy.rst

```diff
@@ -0,0 +1,145 @@
+of game developers, game designers, and they’re tightly integrated in
+the engine and the editor.
+
+|image3|
+
+*VisualScript is a node-based programming language that integrates well
+in the editor. You can drag and drop nodes or resources into the graph
+to create new code blocks.*
+
+Note that if this is true for 2d at the time of writing, the 3d
+workspace still doesn’t feature as many tools. You’ll need external
+programs or add-ons to edit terrains, animate complex characters, etc.
+Godot provides a complete API to extend the editor’s functionality using
+game code. See `The Godot editor is a Godot game <#>`__ below.
+
+|image4|
+
+*A State Machine editor plugin in Godot 2 by kubecz3k. It lets you
+manage states and transitions visually*
+
+Open source
+-----------
+
+Godot offers a fully open source code-base under the MIT license. This
+means all the technologies that ship with it have to be Free as well.
+For the most part, they’re coded from the ground up by contributors.
+
+Anyone can plug in proprietary tools for the needs of their projects.
+They just won’t ship with the engine. This may include NViDia PhysX,
+Google Admob or an FBX file importer. Any of these can come as
+third-party plugins instead.
+
+On the other hand, an open codebase means you can learn from and extend
+the engine to your heart’s content.
+
+Note this does not affect the work you do with Godot in any way: there’s
+no strings attached to the engine or anything you make with it.
+
+Community-driven
+----------------
+
+**Godot is made by its community, for the community and all game
+creators out there**. It’s the needs of the users and open discussions
+that drive the core updates. New features from the core developers often
+focus on what will benefit the most users first.
+
+That said, although a handful of core developers work on it full-time,
+the project has over 500 contributors at the time of writing. Benevolent
+programmers work on features they may need themselves so you’ll see
+improvements in all corners of the engine at the same time in every
+major release.
+
+The Godot editor is a Godot game
+--------------------------------
+
+The Godot editor runs on the game engine. It uses the engine’s own UI
+system, it can hot reload code and scenes when you test your projects,
+or run game code in the editor. This means you can **use the same code**
+and scenes for your games or **to build plugins and extend the editor**.
+
+This leads to a reliable and flexible UI system as it powers the editor
+itself.
+
+|image5|
+
+*RPG in a Box is a voxel RPG editor made in Godot 2. It uses Godot’s UI
+tools for its node-based programming system and for the rest of the
+interface.*
+
+Separate 2d and 3d engines
+--------------------------
+
+Godot offers dedicated 2d and 3d rendering engines. As a result **the
+base unit for 2d scenes is pixels**. Even though the engines are
+separate, You can render 2d in 3d, 3d in 2d, and overlay 2d sprites and
```

Comment:
*you* (case), also: not only rendering engines, but physics engines. Also maybe mention new Bullet integration for 3D somewhere